h/static/scripts/directive/excerpt.coffee

```diff
@@ -0,0 +1,35 @@
+###*
+# @ngdoc directive
+# @name excerpt
+# @restrict C
+# @description Checks to see if text is overflowing its container.
+# If so, it prepends/appends expanders/collapsers. Note, to work
+# the element needs a max height.
+###
+module.exports = [ '$timeout', ($timeout) ->
+  link: (scope, elem, attr, ctrl) ->
+    # Check for excerpts if an annotation has been created / edited.
+    scope.$watch (-> scope.vm.editing), (editing) ->
+      if editing
+        elem.find('.more').remove()
+        elem.find('.less').remove()
+      scope.excerpt()
+
+    # Check for excerpts on threadCollapseToggle event.
+    scope.$on 'threadToggleCollapse', (value) ->
+      scope.excerpt()
+
+    do scope.excerpt = ->
```

Comment:
This is fine as far as it goes. It works! But it's not terribly "angulary" -- in that we're attaching our own events and classes manually using jQuery rather than changing some state in the directive scope and having the markup change in the Angular digest cycle.